skiko/src/awtMain/cpp/windows/JumpList.cc

```diff
@@ -0,0 +1,349 @@
+
+    jstring jDescription = (jstring) env->CallObjectMethod(obj, org::jetbrains::skiko::windows::JumpListInteropItem::getDescription);
+    jstring jIconPath = (jstring) env->CallObjectMethod(obj, org::jetbrains::skiko::windows::JumpListInteropItem::getIconPath);
+    int jIconNum = (int) env->CallIntMethod(obj, org::jetbrains::skiko::windows::JumpListInteropItem::getIconNum);
+
+    std::optional<std::wstring> description =
+        env->IsSameObject(jDescription, NULL) ? std::nullopt : std::make_optional(toStdString(env, jDescription));
+    std::optional<std::pair<std::wstring, int>> icon =
+        env->IsSameObject(jIconPath, NULL) ? std::nullopt : std::make_optional(std::make_pair(toStdString(env, jIconPath), jIconNum));
+
+    return createShellLink(env, title, arguments, description, icon, ppsl);
+}
+
+jobject createJumpListInteropItemObject(JNIEnv* env, IShellLinkW *psl)
+{
+    org::jetbrains::skiko::windows::ensure(env);
+
+    ComPtr<IPropertyStore> propStore;
+    if (SUCCEEDED(psl->QueryInterface(IID_PPV_ARGS(&propStore)))) {
+        PROPVARIANT propvar;
+        if (SUCCEEDED(propStore->GetValue(PKEY_Title, &propvar))) {
+            PropVariantWrapper propvarWrapper(propvar);
+            WCHAR szTitle[512];
+            if (SUCCEEDED(PropVariantToString(propvar, szTitle, ARRAYSIZE(szTitle)))) {
+                WCHAR szArguments[1024];
+                if (SUCCEEDED(psl->GetArguments(szArguments, ARRAYSIZE(szArguments)))) {
+                    WCHAR szIconPath[512];
+                    int numIcon;
+                    if (SUCCEEDED(psl->GetIconLocation(szIconPath, ARRAYSIZE(szIconPath), &numIcon))) {
+                        WCHAR szDescription[1024];
+                        if (SUCCEEDED(psl->GetDescription(szDescription, ARRAYSIZE(szDescription)))) {
+                            return env->NewObject(
+                                org::jetbrains::skiko::windows::JumpListInteropItem::cls,
+                                org::jetbrains::skiko::windows::JumpListInteropItem::init,
+                                env->NewString(reinterpret_cast<const jchar*>(szTitle), wcsnlen_s(szTitle, ARRAYSIZE(szTitle))),
+                                env->NewString(reinterpret_cast<const jchar*>(szArguments), wcsnlen_s(szArguments, ARRAYSIZE(szArguments))),
+                                env->NewString(reinterpret_cast<const jchar*>(szDescription), wcsnlen_s(szDescription, ARRAYSIZE(szDescription))),
+                                env->NewString(reinterpret_cast<const jchar*>(szIconPath), wcsnlen_s(szIconPath, ARRAYSIZE(szIconPath))),
+                                numIcon
+                            );
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    return NULL;
+}
+
+extern "C"
+{
+    JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_windows_JumpListBuilder_jumpList_1init(JNIEnv* env, jobject obj) {
```

Comment:
CI fails here:
https://github.com/JetBrains/skiko/actions/runs/9554903474/job/26421023172?pr=943

Reply:
Yes, I forgot to add a return value to the COM initialization success check 😬

Reply:
Done